### PR TITLE
[#3075] fix(core): Fix issues schema entity cannot delete if the schema is not found from underlying sources

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

Ignore the return value of dropSchema from underlying sources, to drop the entity from store mandatorily.

### Why are the changes needed?

If the return value of `dropSchema` is false from underlying sources, the current implementation will ignore deleting it from entity store, so it cannot drop the catalog.

Fix: #3075 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Adding new ITs.

### CODE REVIEW

**Comment by jerryshao** [comment]
@xunliu can you please verify with this fix?

File: core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java [hunk 1]
```diff
@@ -297,18 +298,19 @@ public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
...
+        getCatalogIdentifier(ident),
+        c -> c.doWithSchemaOps(s -> s.dropSchema(ident, cascade)),
+        NonEmptySchemaException.class);
 
+    // It could happen that the schema is not found in the catalog (dropped directly from
+    // underlying sources), but it is still in the store. So we should ignore the return value
+    // from catalog operation and try to delete the schema entity from the store.
     try {
       return store.delete(ident, SCHEMA, cascade);
+    } catch (NoSuchEntityException e) {
```
Comment (yuqi1129) [inline core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java:311]: Is the exception `NoSuchEntityException` thrown by the JDBC backend?

Comment (jerryshao) [reply core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java:311]: Yes.

**Review by yuqi1129 — DISMISSED** [review]
LGTM

**Comment by jerryshao** [comment]
The change still have some issues, continue to work on it.

File: core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java [hunk 2]
```diff
@@ -297,20 +298,35 @@ public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
...
+    //
+    // For managed schema, we should take the return value of the store operation into account.
+    boolean droppedFromStore = false;
     try {
-      return store.delete(ident, SCHEMA, cascade);
+      droppedFromStore = store.delete(ident, SCHEMA, cascade);
+    } catch (NoSuchEntityException e) {
+      LOG.warn("The schema to be dropped does not exist in the store: {}", ident, e);
     } catch (Exception e) {
       throw new RuntimeException(e);
```
Comment (mchades) [inline core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java:325]: For unmanaged schema, should we throw an exception here?

**Review by mchades — COMMENTED** [review]
comment on the API and method should also be modified

File: core/src/main/java/com/datastrato/gravitino/catalog/TableOperationDispatcher.java [hunk 3]
```diff
@@ -300,23 +302,33 @@ public Table alterTable(NameIdentifier ident, TableChange... changes)
...
+    // return value of the store operation into account. We only take the return value of the
+    // catalog into account.
+    //
+    // For managed table, we should take the return value of the store operation into account.
+    boolean droppedFromStore = false;
     try {
-      store.delete(ident, TABLE);
+      droppedFromStore = store.delete(ident, TABLE);
+    } catch (NoSuchEntityException e) {
+      LOG.warn("The table to be dropped does not exist in the store: {}", ident, e);
```
[the remark this answers is not in the document]
Comment (jerryshao) [reply core/src/main/java/com/datastrato/gravitino/catalog/TableOperationDispatcher.java:324]: Fixed.

**Review by mchades — APPROVED** [review]
LGTM

**Pull request merged**